File: python/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for i in range(len(nums)):
          map[nums[i]] = i
        for j in range(len(nums)):
          complement = target - nums[j]
          if complement in map and map[complement] != j:
              return [j, map[complement]]
        return -1

File: cpp/twoSum.cpp
using namespace std; 

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> indexMaps; 
        vector<int> twoSumPair;
        for (int i = 0; i < nums.size(); i++) {
            indexMaps[nums[i]] = i;
        }
        int complement;
        for (int j = 0; j < nums.size(); j++) {
            complement = target - nums[j];
            if (indexMaps.count(complement) && j != indexMaps[complement]) {
                twoSumPair.push_back(j);
                twoSumPair.push_back(indexMaps[complement]);
                return twoSumPair;
            }
        }
      return twoSumPair;
    }
};
